src/api.rs

```diff
@@ -76,8 +76,15 @@ pub struct EncoderConfig {
   pub quantizer: usize,
   pub bitrate: i32,
   pub tune: Tune,
+  /// log2(tile columns). If tiles is also specified, this acts
+  /// as a minimum.
   pub tile_cols_log2: usize,
+  /// log2(tile rows). If tiles is also specified, this acts
+  /// as a minimum.
   pub tile_rows_log2: usize,
+  /// Number of tiles desired. The video is split automatically so that
+  /// it contains at least this many tiles.
```

Comment:
I don't know why, but "at most" this many tiles would be less confusing. It seems weird to request 10 tiles and end up with 16.

Moreover, on small video (for example 3×3 SB), if you request 10 tiles, you'll get 9. (OK, this is a weak argument, you'll have the symmetrical behavior with big videos).

Reply:
It is hard to tell which would be cleaner. If returning an error is a viable option I think I would prefer that over automatically "fixing" the number of tiles.

Reply:
Returning an error is a viable option. The only downside is that it's not always obvious what #s of tiles are allowed